src/cpp/session/modules/SessionDataImportV2.R

```diff
@@ -255,29 +259,51 @@
       cacheDataCode <- list()
 
       localFile <- unlist(options$localFiles[[resource]])
+      canCacheData <- identical(options$canCacheData, TRUE)
+      cacheDataWorkingDir <- !identical(options$cacheDataWorkingDir, NULL) && options$cacheDataWorkingDir == TRUE
 
-      if (importFromUrl)
+      if (importFromUrl && canCacheData)
       {
-         if (identical(localFile, NULL))
+         cacheVariableName <- options$cacheVariableNames[[resource]]
+
+         if (cacheDataWorkingDir)
          {
-            localFile <- tempfile(
-               tmpdir = dirname(tempdir()),
-               fileext = functionInfo$cacheFileExtension
+            cacheDataCode <- append(
+               cacheDataCode,
+               paste(
+                  cacheVariableName,
+                  " <- tempfile(tmpdir = getwd(), fileext = \"",
```

Comment:
I wonder if it would be nicer to just pick the filename out of the URL and then use that rather than a temp name (it would certainly be much easier later on to sort out what the file is if it retains it's original filename)